### DISCUSSION TITLE
Trigger rule on Timers change

### DISCUSSION THREAD

**Discussion by nonix** [arendst/Tasmota#11612] [Q&A] [answered]
Is it possible to trigger a rule when Global timers are enabled/disabled?

WebLog shows:
_17:59:01.562 MQT: stat/clock/RESULT = {"Timers":"ON"}_

I have tried something like: _on Timers#Data=ON do event TON endon_but it won't fire.

Also, when timers are change via WebUI it won't send the MQT:stat ...  Is this by intention?

Thank you,
  N.

**Comment by barbudor** [comment, marked as answer]
Unfortunately `Timers` is one of the few commands which response doesn't go through the Rules engine
This would requires a change in the code

**Comment by nonix** [comment]
Yah I thought so,
Thank you for answering.



N.

**Reply by barbudor** [reply]
How is that critical for you ?

**Reply by nonix** [reply]
Hi barbudor, 'm sorry did not mean to be rude in lights of [comment from sfromis](https://github.com/arendst/Tasmota/discussions/11612?_pjax=%23js-repo-pjax-container#discussioncomment-584281)  I did not know you are actually working on it. Many thanks for support :-)

**Reply by barbudor** [reply]
No offense taken. I didn't found that rude :)
And I did nothing except reporting your issue to Theo who decided to do the actual change.
Enjoy Tasmota !

**Comment by nonix** [comment]
Not a drama, and not a high priority. 'm working on micro Easter project an alarm clock :-) and would like to toggle LED when timers are toggled (On or Off). 

I will try to dig into the code. If I succeed, will post a PR for the change.

Thank you,
 N.

**Reply by sfromis** [reply]
Depending on details, you could adjust a bit to have a rule like `on Timer1#Enable do ....` and of course enable/disable that timer. The `Timer` command output is available to rule triggers.

**Reply by sfromis** [reply]
With latest [dev binary](http://ota.tasmota.com/tasmota/) `on Timers#data do var1 %value% endon` now works

**Reply by nonix** [reply]
Hi sfromis, That's very nice of you, thank you. I can confirm it now works, when Timers command is fired via MQTT/cons.
However, when the status is changed via Web UI it does not fire.

I am grateful for the change as is now, however since you are more skilled in tasmota than myself and if your time permits, it would great to have the status changed noted also via WebUI.

Thanks a lot for helping,
 N.

**Reply by sfromis** [reply]
Well, I did not do anything, I just mentioned a result of barbudor looking into the case. Rule triggers work based on the JSON payloads published via MQTT, but some payloads were not processed by rules. Now additional ones are. As the timers web UI for some reason does not publish JSON payloads, they are not available as things are now. The debugging format which printed in the console is outside of what triggers have access to.
`Timers 1,0x00000000,0x187F035E,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000`

**Comment by arendst** [comment]
Leave it with me. I'll just add an execute the ``timers`` command from the GUI et presto.

**Reply by sfromis** [reply]
Works. Now also using the dialog will result in the JSON payloads picked up by rules.

**Comment by arendst** [comment]
Fixed in latest development version

**Reply by nonix** [reply]
The Webi now works, great support.

It seems I have one more problem, but not sure if it related or just my misunderstanding.

on Button1#Action=DOUBLE do Timers 2 endon   // Toggle
on Timers#Data=ON do power2 on endon
on Timers#Data=OFF do power2 off endon

When the "Timers cmd" is send by MQTT/Cons/Webi => all works as expected.
When I press the button, the firs rule fires OK, but the no change on Power :-( 

Can you advice, please.

Thx a lot,
 N.

**Reply by sfromis** [reply]
That's another niggle. To reliably have the output of commands from rules parsed by other rule triggers, you need to wrap it in Backlog. Try `on Button1#Action=DOUBLE do Backlog Timers 2 endon`

**Reply by nonix** [reply]
Guys, thanks a lot for all of you great help and support provided. Tasmota rocks!
Cheers,
 N.
